android/src/main/java/com/aerobotics/DjiMobile/DJIMobile.java

```diff
@@ -530,6 +537,34 @@ public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue)
     });
   }
 
+  private void startDiagnosticsListener() {
+    // Add product connection listener
+    DJIDiagnostics.DiagnosticsInformationCallback diagnosticsInformationCallback = new DJIDiagnostics.DiagnosticsInformationCallback() {
+      @Override
+      public void onUpdate(List<DJIDiagnostics> list) {
+        if (!list.isEmpty()) {
+          WritableArray diagnosticsToSend = Arguments.createArray();
+          for (DJIDiagnostics djiDiagnostics : list) {
+            DiagnosticsBaseHandler.DJIDiagnosticsError error = DiagnosticsBaseHandler.DJIDiagnosticsError.find(djiDiagnostics.getCode());
+            WritableMap params = Arguments.createMap();
+            params.putString("type", djiDiagnostics.getType().name());
+            params.putString("reason", djiDiagnostics.getReason());
+            params.putString("solution", djiDiagnostics.getSolution());
+            params.putString("error", error.name());
```

Comment:
What is the value of this when there is no error?

Reply:
If there's no error then the list is empty